curp/src/members.rs

```diff
@@ -0,0 +1,200 @@
+use std::{
+    collections::{hash_map::DefaultHasher, HashMap},
+    hash::Hasher,
+    time::SystemTime,
+};
+
+use itertools::Itertools;
+
+use crate::ServerId;
+
+/// cluster members information
+#[derive(Debug)]
+pub struct ClusterMember {
+    /// current server id
+    id: String,
+    /// other peers information
+    all_members: HashMap<ServerId, String>,
+}
+
+impl ClusterMember {
```

Comment:
Like what?

Reply:
For instances, we can store self address to the `ClusterMember` struct to avoid lookup every time in `get_self_url`.  And we can also store the return value of `get_peers_id` in the struct to avoid calculation.

Reply:
Done